bin/callback.js

```diff
@@ -0,0 +1,84 @@
+const http = require('http')
+
+const CALLBACK_URL = process.env.CALLBACK_URL ? new URL(process.env.CALLBACK_URL) : null
+const CALLBACK_TIMEOUT = process.env.CALLBACK_TIMEOUT || 5000
+const CALLBACK_OBJECTS = process.env.CALLBACK_OBJECTS ? JSON.parse(process.env.CALLBACK_OBJECTS) : {}
+
+/**
+ * @return {Boolean}
+ */
+exports.isCallbackSet = () => {
+  if (CALLBACK_URL) {
+    return true
+  }
+  return false
+}
+
+/**
+ * @param {Uint8Array} update
+ * @param {any} origin
+ * @param {WSSharedDoc} doc
+ */
+exports.callbackHandler = (update, origin, doc) => {
+  const room = doc.name
+  const dataToSend = {
+    room: room,
+    data: {}
+  }
+  const sharedObjectList = Object.keys(CALLBACK_OBJECTS)
+  sharedObjectList.forEach(sharedObjectName => {
+    const sharedObjectType = CALLBACK_OBJECTS[sharedObjectName]
+    dataToSend.data[sharedObjectName] = {
+      type: sharedObjectType,
+      content: getContent(sharedObjectName, sharedObjectType, doc)
+    }
+  })
+  callbackRequest(CALLBACK_URL, CALLBACK_TIMEOUT, dataToSend)
+}
+
+/**
+ * @param {URL} url
+ * @param {number} timeout
+ * @param {Object} data
+ */
+const callbackRequest = (url, timeout, data) => {
+  data = JSON.stringify(data)
```

Comment:
This will probably call `toString` on the Yjs type. I think it would be better if the content returned is `type.toJSON()`

Reply:
`data` now receives serialized data. So `JSON.stringify(data)` will now stringify the serialized data and work as expected.